### PR TITLE
Storybook: Fix missing `favorite` and `grafana` icons

### LINKED ISSUE #90374: Storybook: `favorite` and `grafana` icons are not shown

### PR DESCRIPTION
**Pull request by ashharrison90**
**What is this feature?**

- improves the icon filtering logic to only filter out the `fa fa-spinner` icon

**Why do we need this feature?**

- so `favorite` and `grafana` icons are displayed properly in storybook

**Who is this feature for?**

- everyone! 🙌 

**Which issue(s) does this PR fix?**:

Fixes https://github.com/grafana/grafana/issues/90374

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

**Review by JoaoSilvaGrafana — APPROVED** [review]
Lol.

File: packages/grafana-ui/.storybook/main.ts [hunk 1]
```diff
@@ -15,7 +15,7 @@ const stories = ['../src/Intro.mdx', storyGlob];
 // We limit icon paths to only the available icons so publishing
 // doesn't require uploading 1000s of unused assets.
 const iconPaths = Object.keys(availableIconsIndex)
-  .filter((iconName) => !iconName.includes('fa'))
+  .filter((iconName) => !iconName.startsWith('fa '))
```
Comment (eledobleefe) [inline packages/grafana-ui/.storybook/main.ts:18]: I can see the icons now, but will this still block the icons we want to avoid? 🤔

Comment (ashharrison90) [reply packages/grafana-ui/.storybook/main.ts:18]: yes, the one that wants to be avoided is `fa fa-spinner`

**Review by eledobleefe — APPROVED** [review]

File: packages/grafana-ui/.storybook/main.ts (hunk 1 above)
Comment (eledobleefe) [reply packages/grafana-ui/.storybook/main.ts:18]: Yes, that's right. I was thinking it was `fa-spinner`, without the previous `fa` 😅

**Pull request merged**